File: src/Services/DynamicPageController.ts

import * as common from "../Views/common";
import { Result } from "../Utils/TTTServiceResult";
import { Language } from "../Views/common";

export async function get(p:{ itemCode:string, language?:Language }) : Promise<string>
{
	if( p.language == null )
		p.language = common.pageParameters.currentLanguage;
	const rc = <Result>await common.url.postRequestForm( common.routes.api.dynamicPage.get, p );
	return rc.result;
}

export async function update(p:{ itemCode:string, language?:Language, text:string, allowAdd?:boolean }) : Promise<string>
{
	if( p.language == null )
		p.language = common.pageParameters.currentLanguage;
	const rc = <Result>await common.url.postRequestForm( common.routes.api.dynamicPage.update, p );
	return rc.result;
}

File: app/src/Views/DynamicPage/DynamicPageController.cs

using System.Threading.Tasks;

using Microsoft.AspNetCore.Mvc;

namespace ItemTTT.Views
{
	public class DynamicPageController : BaseController
	{
		private const string	ShowView	= Startup.ViewsLocation+"DynamicPage/Show.cshtml";

		private readonly PageHelper				PageHelper;
		private readonly Models.ItemTTTContext	DataContext;

		public DynamicPageController(Models.ItemTTTContext dataContext, PageHelper pageHelper)
		{
			PageHelper = pageHelper;
			DataContext = dataContext;
		}

		[HttpGet( Routes.DynamicPageStatic )]
		public Task<IActionResult> StaticDemo()
		{
			return Show( lang:PageHelper.CurrentLanguage, itemCode:"static-demo" );
		}

		[HttpGet( Routes.DynamicPageShow )]
		public async Task<IActionResult> Show(Languages lang, string itemCode)
		{
			var rv = await (new Services.DynamicPageController( DataContext, PageHelper )).Get( itemCode:itemCode, language:lang );
			if(! rv.Success )
				return ObjectNotFound();

			var model = new ShowModel {	ItemCode	= itemCode,
										Text		= rv.Result };
			return View( ShowView, model );
		}
		public struct ShowModel
		{
			public string	ItemCode;
			public string	Text;
		}
	}
}
